pyiron_contrib/generic/s3io.py

```diff
@@ -0,0 +1,564 @@
+import fnmatch
+import io
+import json
+import os
+import posixpath
+from functools import lru_cache
+
+import boto3
+from botocore.client import Config
+
+from pyiron_base.generic.filedata import load_file, FileDataTemplate
+
+
+class S3FileData(FileDataTemplate):
+    """FileData stores an instance of a data file from an S3 system, e.g. a single Image from a measurement."""
+    def __init__(self, s3obj, filename=None, filetype=None):
+        """FileData class to store data and associated metadata.
+
+            Args:
+                s3obj (s3object): s3object containing a file with data
+                filename (str): filename associated with the data.
+                filetype (str): File extension associated with the type data,
+                                If provided this overwrites the assumption based on the extension of the filename.
+        """
+        self._s3obj = s3obj
+        if filename is None:
+            self.filename = s3obj.key.split('/')[-1]
+        else:
+            self.filename = filename
+        if filetype is None:
+            filetype = os.path.splitext(self.filename)[1]
+            if filetype == '' or filetype == '.':
+                self.filetype = None
+            else:
+                self.filetype = filetype[1:]
+        else:
+            self.filetype = filetype
+        self._data = None
+        self._metadata = self._s3obj.metadata
+
+    @property
+    @lru_cache()
+    def data(self):
+        """Return the associated data."""
+        if self._data is None:
+            self._s3obj = self._s3obj.get()
+            self._data = self._s3obj["Body"].read()
+        return load_file(io.BytesIO(self._data), filetype=self.filetype)
+
+    @property
+    def metadata(self):
+        return self._metadata
+
+    def __repr__(self):
+        return f"FileData containing {self.filename}."
+
+    def _ipython_display_(self):
+        result = f"File    {self.filename}   with meta data:\n"
+        result += "-------------------------------------------\n"
+        for key, value in self.metadata.items():
+            result += f"{key}:  {value}\n"
+        print(result)
+
+
+class S3ioConnect:
+    def __init__(self, credentials=None, bucket_name=None):
```

Comment:
This is a bit hard to parse for the newcomer -- can you refactor some of this behaviour out into clearly named methods? I am imagining that they would be simple, only a two or three lines, but giving a name to the behaviour will help readability.